app/src/main/java/ch/epfl/sweng/SDP/Activity.java

```diff
@@ -139,11 +139,12 @@ protected void cloneAccountFromFirebase(@NonNull DataSnapshot snapshot) {
                         ((Long) user.get("stars")).intValue(),
                         ((Long) user.get("matchesWon")).intValue(),
                         ((Long) user.get("totalMatches")).intValue(),
-                        ((Long) user.get("averageRating")).doubleValue(),
+                        (Long) user.get("averageRating"),
```

Comment:
Does it work? Because Account expects a double as averageRating

Reply:
It works if your actual ranking is a long (i.e 6.0) otherwise it needs to be casted as a double. The problem is that we cannot cast this into double because if we get a long number it won't work but for other cases yes. We need to find some way to find the type of the value before casting it.